src/meta/TableSpec.h

```diff
@@ -185,5 +202,52 @@ struct TableSpecEqual {
 
 using TableSpecSet = nebula::common::unordered_set<TableSpecPtr, TableSpecHash, TableSpecEqual>;
 
+constexpr auto HOUR_MINUTES = 60;
+constexpr auto MINUTE_SECONDS = 60;
+constexpr auto DAY_HOURS = 24;
+constexpr auto HOUR_SECONDS = HOUR_MINUTES * MINUTE_SECONDS;
+constexpr auto DAY_SECONDS = HOUR_SECONDS * DAY_HOURS;
+
+const static nebula::common::unordered_map<nebula::meta::PatternMacro, std::string> patternStr{ { nebula::meta::PatternMacro::DATE, "date" }, { nebula::meta::PatternMacro::HOUR, "hour" }, { nebula::meta::PatternMacro::MINUTE, "minute" }, { nebula::meta::PatternMacro::SECOND, "second" }, { nebula::meta::PatternMacro::TIMESTAMP, "timestamp" } };
+const static nebula::common::unordered_map<nebula::meta::PatternMacro, nebula::meta::PatternMacro> childPattern{ { nebula::meta::PatternMacro::DATE, nebula::meta::PatternMacro::HOUR }, { nebula::meta::PatternMacro::HOUR, nebula::meta::PatternMacro::MINUTE }, { nebula::meta::PatternMacro::MINUTE, nebula::meta::PatternMacro::SECOND } };
+const static nebula::common::unordered_map<nebula::meta::PatternMacro, int> partitionInSeconds{ { nebula::meta::PatternMacro::DATE, DAY_SECONDS }, { nebula::meta::PatternMacro::HOUR, HOUR_SECONDS }, { nebula::meta::PatternMacro::MINUTE, MINUTE_SECONDS } };
+const static nebula::common::unordered_map<nebula::meta::PatternMacro, int> partitionSize{ { nebula::meta::PatternMacro::DATE, DAY_HOURS }, { nebula::meta::PatternMacro::HOUR, HOUR_MINUTES }, { nebula::meta::PatternMacro::MINUTE, MINUTE_SECONDS } };
+
+// check if pattern string type
+inline nebula::meta::PatternMacro extractPatternMacro(std::string pattern) {
+  // ts=?
+  if (pattern.find(patternStr.at(nebula::meta::PatternMacro::TIMESTAMP))) {
+    if (!pattern.find(patternStr.at(nebula::meta::PatternMacro::DATE)) && !pattern.find(patternStr.at(nebula::meta::PatternMacro::HOUR)) && !pattern.find(patternStr.at(nebula::meta::PatternMacro::MINUTE)) && !pattern.find(patternStr.at(nebula::meta::PatternMacro::SECOND))) {
```

Comment:
to prevent repeatedly `find` calls, can we introduce local variables for them? 
```
const auto tsMacroFound = pattern.find(MacroPattern.at(PatternMacro::TIMESTAMP)) != std::string::npos;
const auto dateMacroFound = pattern.find(MacroPattern.at(PatternMacro::DATE)) != std::string::npos;
const auto hourMacroFound = pattern.find(MacroPattern.at(PatternMacro::HOUR)) != std::string::npos;
const auto minuteMacroFound = pattern.find(MacroPattern.at(PatternMacro::MINUTE)) != std::string::npos;
...
if(xxFond){
 ..
}
else if(yyFound && ...){
  ...
}
```

This will make the code a bit more readable.